tests/composition/test_show_graph.py

```diff
@@ -233,28 +234,83 @@ def test_converging_pathways(self):
+        )
+    ]
 
+    @pytest.fixture(scope='session')
+    def _no_nested_and_controler_name_with_space_in_it_comp(self):
```

Comment:
This is a rather hacky missuse of fixtures. The function is called only in one place and there's no reason for it to be a fixture or a standalone function.
Making this a fixture moves the composition construction outside of the test with hard to read/diagnose error messaging if it fails.

Reply:
The main reason was that each parametrization runs the composition setup again, which caused the set of show_graph tests to run noticeably slower, at least locally. Using a scoped fixture avoids this, although depending on how much it affects GHA/jenkins (will check) it may not be worth worrying about. I admit it's ugly though.

Reply:
There's enough variance in the duration of the test suite from run to run that it's not clear how much time the fixtures save, so I've removed them.

Reply:
It's probably not the case with show_graph, but rebuilding the complete setup for each test should help avoid transitive/cascading failures.

Reply:
another option would be add a test class and have the constructed composition be a class member accessible in every test function. I think that would also match the show_graph usage pattern.

Reply:
For now I'll leave it as rebuilding - there aren't too many show_graph tests and I don't expect many more will be added - but that's an option if we decide we need to reduce the test suite length in the future.